runtime/include/localeModels/flat/chpl-locale-model.h

```diff
@@ -66,4 +66,17 @@ c_sublocid_t chpl_rt_sublocFromLocaleID(chpl_localeID_t loc) {
   return c_sublocid_any;
 }
 
+//
+// These functions are exported from the locale model for use by
+// the tasking layer to convert between a full sublocale and an
+// execution sublocale.
+//
+extern
+c_sublocid_t chpl_localeModel_sublocToExecutionSubloc(
```

Comment:
The problem is that it's the tasking layer that caches the requested sublocale via chpl_task_setSubloc and retrieves it with chpl_task_getRequestedSubloc.  To also store it in the locale model, we would need to cache in two places, and it would be hard to ensure that no future changes called the wrong layer.